src/fstext/fstext-utils-inl.h

```diff
@@ -114,6 +114,52 @@ void GetInputSymbols(const Fst<Arc> &fst,
   std::sort(symbols->begin(), symbols->end());
 }
 
+template<class Arc, class I>
+class RemoveSomeInputSymbolsMapper {
+public:
+  Arc operator ()(const Arc &arc_in) {
+    Arc ans = arc_in;
+    if (to_remove_set_.count(ans.ilabel) != 0) ans.ilabel = 0;  // remove this symbol
+    return ans;
+  }
+  MapFinalAction FinalAction() { return MAP_NO_SUPERFINAL; }
+  MapSymbolsAction InputSymbolsAction() { return MAP_CLEAR_SYMBOLS; }
+  MapSymbolsAction OutputSymbolsAction() { return MAP_COPY_SYMBOLS; }
+  uint64 Properties(uint64 props) const {
+    // remove the following as we don't know now if any of them are true.
+    uint64 to_remove = kAcceptor|kNotAcceptor|kIDeterministic|kNonIDeterministic|
+        kNoEpsilons|kNoIEpsilons|kILabelSorted|kNotILabelSorted;
+    return props & ~to_remove;
+  }
+  RemoveSomeInputSymbolsMapper(const std::vector<I> &to_remove):
+      to_remove_set_(to_remove) {
+    KALDI_ASSERT_IS_INTEGER_TYPE(I);
+         assert(to_remove_set_.count(0) == 0);  // makes no sense to remove epsilon.
+       }
+private:
+  kaldi::ConstIntegerSet<I> to_remove_set_;
+};
+
+// Lookahead composition is used for optimized online
+// composition of FSTs during decoding. See
+// nnet3/nnet3-latgen-faster-lookahead.cc
+template<class Arc, class I>
+ArcMapFst<Arc, Arc, RemoveSomeInputSymbolsMapper<Arc, I> > LookaheadComposeFst(const Fst<Arc> &ifst1,
```

Comment:
Not sure how to simplify it, need help from C++ expert on how to specialize a template.

Reply:
Why do you want to specialize this template function?

Reply:
Too long probably, could be something like 

```
template<class Arc, class I>
ComposedNodisambigFst<Arc, I> LookaheadComposeFst(const Fst<Arc> &ifst1
```

Reply:
```
template<class Arc, class I>
using ComposedNodisambigFst = ArcMapFst<Arc, Arc, RemoveSomeInputSymbolsMapper<Arc, I> > 
```

Reply:
Very good, thank you!